ingester/util/ticker.go

```diff
@@ -0,0 +1,57 @@
+package util
+
+import (
+	"time"
+
+	"chainlink/ingester/logger"
+)
+
+// TickerService is an interface that provides
+// basic ticking functionality to a service
+type TickerService interface {
+	Start()
+	Stop()
+	Tick()
+}
+
+// Ticker is the implementation of the TickerService
+// interface, managing the lifecycle of a ticking service
+type Ticker struct {
+	Ticker *time.Ticker
+	Name   string
+	Impl   TickerService
+	Done   chan struct{}
+	Exited chan struct{}
+}
+
+// Start will start the ticker service
+func (t *Ticker) Start() {
+	go t.tick()
+}
+
+// Stop will stop the ticket service, waiting
+// for any any tick to first complete
+func (t *Ticker) Stop() {
+	close(t.Done)
+	logger.Info("Waiting for service to exit")
```

Comment:
Unless you use something like `_, fn, line, _ := runtime.Caller(1)` this is going to look like:

```
Waiting for service to exit
Waiting for service to exit
Waiting for service to exit
Waiting for service to exit
```

when my program exits, which doesn't feel super helpful.